tests/test_readme.py

```diff
@@ -0,0 +1,59 @@
+import pathlib
+import re
+import subprocess
+
+import pytest
+
+readme_path = pathlib.Path(__file__).parent.parent / 'README.md'
+readme = readme_path.read_text()
+
+template = r'''
+<!-- test codeblock contains output of `{command}` -->
+```
+{output}```
+'''
+pattern = template.format(
+    command=r"(?P<command>.+?)",
+    output=r"(?P<output>.+?)",
+)
+pattern = re.compile(pattern, re.DOTALL)
+matches = list(pattern.finditer(readme))
+
+
+@pytest.mark.parametrize(
+    argnames=['command', 'expected'],
+    argvalues=[match.groups() for match in matches],
+    ids=[match.group('command') for match in matches],
+)
+def test_readme_codeblock_contains_output_from(command, expected):
+    """
+    If this test fails, ensure that codeblocks in README.md have the correct
+    output. To enable this check for output in a codeblock, use the following
+    construct:
+
+    <!-- test codeblock contains output of `{command}` -->
+    ```
+    {expected}
+    ```
+    """
+    output = _get_output_from(command)
+    assert output == expected
+
+
+def _get_output_from(command):
+    return subprocess.check_output(command, shell=True, universal_newlines=True)
```

Comment:
Maybe add some comments to describe what each function is testing for

Reply:
This is just a helper function. The tests all begin with `test_` (pytest is smart like that)!

This function is just a wrapper around subprocess.check_output, so we don't have to repeat the command / arguments twice in two parts of this file.

Will take a look to make sure all tests have docstrings that this PR adds.